packages/firebase_auth/firebase_auth_platform_interface/lib/src/method_channel/method_channel_firebase_auth.dart

```diff
@@ -604,23 +527,60 @@ class MethodChannelFirebaseAuth extends FirebaseAuthPlatform {
       );
     }
 
-    int handle = MethodChannelFirebaseAuth.nextMethodChannelHandleId;
-
-    _phoneAuthCallbacks[handle] = PhoneAuthCallbacks(verificationCompleted,
-        verificationFailed, codeSent, codeAutoRetrievalTimeout);
-
     try {
-      await channel
-          .invokeMethod<void>('Auth#verifyPhoneNumber', <String, dynamic>{
+      final eventChannelName = await channel
+          .invokeMethod<String>('Auth#verifyPhoneNumber', <String, dynamic>{
         'appName': app.name,
-        'handle': handle,
         'phoneNumber': phoneNumber,
         'timeout': timeout.inMilliseconds,
         'forceResendingToken': forceResendingToken,
         'autoRetrievedSmsCodeForTesting': autoRetrievedSmsCodeForTesting,
       });
+
+      EventChannel(eventChannelName!)
+          .receiveBroadcastStream()
+          .listen((arguments) {
+        final name = arguments['name'];
+        if (name == 'Auth#phoneVerificationCompleted') {
+          final int token = arguments['token'];
+          final String /*!*/ smsCode = arguments['smsCode'];
```

Comment:
auth is already null safe. There's no need for such comment

Does this implies that `token` and other variables **without** this comment should be nullable instead?
Due to implicit-cast being currently enable, that may be a source of a bug

Reply:
`token` should be NOT nullable, `smsCode` may be null - just checked in the Android side. correction on the way.